fauna/client/client.py

```diff
@@ -375,6 +379,40 @@ def _query(
           schema_version=schema_version,
       )
 
+  def stream(self, fql: StreamToken | Query) -> Iterator[Any]:
+    return RetryStreamIter(self, fql)
+
+  @contextmanager
+  def _stream(self, fql: StreamToken | Query) -> Iterator[Any]:
+    headers = self._headers.copy()
+    headers[_Header.Format] = "tagged"
+    headers[_Header.Authorization] = self._auth.bearer()
+
+    if not isinstance(fql, StreamToken):
+      token = self.query(fql).data
```

Comment:
Q: how is type checking working in python these days? Can we trust that if not a stream token we'll surely get an instance of a `Query`?